scalding-core/src/main/scala/com/twitter/scalding/typed/cascading_backend/AsyncFlowDefRunner.scala

```diff
@@ -69,6 +71,12 @@ object AsyncFlowDefRunner {
       }
     }
   }
+
+  case class FlowCancellationHandler(flow: Promise[Flow[_]]) extends CancellationHandler {
+    def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit] = {
+      flow.future.map(_.stop)
```

Comment:
I think you actually want to use `onComplete` rather than `map` because `.stop` is a side effect. But maybe scala futures aren't so uptight.

Also, I bet `stop` can throw an exception. What should we do then? I don't think we should fail the future because it could fail other calls to cancel. i think we want something like:
`flow.future.onComplete { flow => Try(flow.stop()) }`

Reply:
👍